platforms/jvm/plugins-application/src/main/java/org/gradle/api/plugins/internal/DefaultJavaApplication.java

```diff
@@ -52,36 +55,36 @@ public Property<String> getMainModule() {
 
     @Override
     public Property<String> getMainClass() {
-        return mainClass;
+        return mainClassName;
```

Comment:
Can we name this consistently with the method name? Or better yet instantiate this as a managed property with an abstract getter?